fix: login and signup component unit tests

kgajera opened the pull request:
This fixes the login and signup component unit tests.

## Changes

The tests were failing because:
- The `prefetch` function on our mock router needed to return a promise
- The `@testing-library` dependencies required updating

## Checklist

- [x] Requires dependency update?
- [x] Generating a new app works

cmejet commented on packages/create-bison-app/template/tests/utils.tsx:
```diff
@@ -80,8 +80,8 @@ const mockRouter: NextRouter = {
   back: jest.fn(),
-  prefetch: jest.fn(),
   beforePopState: jest.fn(),
+  prefetch: jest.fn(() => Promise.resolve()),
```
🎉

cmejet reviewed (changes requested):
This is awesome!
Given the other PR was merged that skipped those tests we will want to make sure we rebase from canary before we merge this but can't wait to have all the tests working again.

kgajera commented:
I just rebased the PR and updated the tests to not be skipped.

![Screen Shot 2021-06-18 at 9 40 35 AM](https://user-images.githubusercontent.com/1087679/122571360-a7ed3580-d01a-11eb-8472-21b26d17dbaa.png)

cmejet reviewed (approved):
![image](https://user-images.githubusercontent.com/7119624/122571529-abc08e00-d001-11eb-99df-a840727469fb.png)
🎉
